bin/external_lib_deps.ml

```diff
@@ -1,188 +1,28 @@
-            |> List.map ~f:Package.Name.to_string
-          in
-          User_message.prerr
-            (User_error.make
-               [ Pp.textf
-                   "The following libraries are missing in the %s context:"
-                   (Dune_engine.Context_name.to_string context_name)
-               ; pp_external_libs missing
-               ]
-               ~hints:
-                 [ Dune_engine.Utils.pp_command_hint
-                     (opam_install_command ?switch_name required_package_names)
-                 ]);
-          true
-      ) else if sexp then (
-        if not by_dir then
-          User_error.raise [ Pp.textf "--sexp requires --unstable-by-dir" ];
-        let lib_deps_by_dir =
-          lib_deps_by_dir
-          |> Path.Source.Map.map ~f:(Lib_name.Map.filteri ~f:is_external)
-          |> Path.Source.Map.filter ~f:(fun m -> not (Lib_name.Map.is_empty m))
-        in
-        let sexp =
-          Path.Source.Map.to_dyn Lib_deps_info.to_dyn lib_deps_by_dir
-          |> Sexp.of_dyn
-        in
-        Format.printf "%a@." Pp.to_fmt
-          (Sexp.pp
-             (List
-                [ Atom (Dune_engine.Context_name.to_string context_name); sexp ]));
-        acc
-      ) else (
-        if by_dir then
-          User_error.raise
-            [ Pp.textf "--unstable-by-dir cannot be used without --sexp" ];
-        User_message.print
-          (User_message.make
-             [ Pp.textf
-                 "These are the external library dependencies in the %s \
-                  context:"
-                 (Dune_engine.Context_name.to_string context_name)
-             ; pp_external_libs externals
-             ]);
-        acc
-      ))
-
 let term =
-  let+ common = Common.term
-  and+ only_missing =
-    Arg.(
-      value & flag
-      & info [ "missing" ] ~doc:{|Only print out missing dependencies|})
-  and+ targets = Arg.(non_empty & pos_all dep [] & Arg.info [] ~docv:"TARGET")
-  and+ by_dir =
-    Arg.(
-      value & flag
-      & info [ "unstable-by-dir" ]
-          ~doc:
-            {|Print dependencies per directory
-                    (this feature is currently unstable)|})
-  and+ sexp =
-    Arg.(value & flag & info [ "sexp" ] ~doc:{|Produce a s-expression output|})
-  in
-  Common.set_common common ~targets:[] ~external_lib_deps_mode:true;
-  let setup, lib_deps =
-    Scheduler.go ~common (fun () ->
-        let open Fiber.O in
-        let+ setup = Memo.Build.run (Import.Main.setup common) in
-        let targets = Target.resolve_targets_exn common setup targets in
-        let request = Target.request targets in
-        let deps = all_lib_deps ~request in
-        (setup, deps))
-  in
-  let failure = run ~by_dir ~setup ~lib_deps ~sexp ~only_missing in
-  if failure then raise Dune_util.Report_error.Already_reported
+  Term.ret
+  @@ let+ _ = Common.term
+     and+ _ = Arg.(value & flag & info [ "missing" ] ~doc:{|unused|})
```

Comment:
Why did you decide to keep some of the flags?

Reply:
So that existing invocation of `dune external-lib-deps` fail with the error message saying that the command was deleted rather than an error message saying the command line argument is unknown.

Reply:
Got it, makes sense.